[SERV-532] Retry Slack API requests on failure

markmatney opened the pull request:
To test:

1. Run `mvn clean test -Dtest=SlackMessageVerticleTest`; observe test success after around 2 seconds (give or take)
2. Disconnect your computer running Maven from the internet **and** repeat step 1; observe test failure after around 30 seconds
3. Repeat step 1 **and** around 15 seconds into the test execution, reconnect your computer to the internet; observe test success after around 20 seconds

Commit:
Bump pinned OS dependency versions

Commit:
Factor out the API response checking

Commit:
Use request builder syntax

Commit:
Move exception-throwing calls outside of try-catch

Commit:
Group imports

Commit:
Use simpler List creation syntax

Commit:
Variable names and line wrapping

Commit:
Retry sending Slack API requests on failure

cachemeoutside reviewed (dismissed):
I was able to follow the code through my non-java expertise. Looks good to me.

DRickard reviewed (dismissed):
Ran tests and got results predicted in PR, code looks good.

Commit:
Adjust event bus timeouts to accommodate Slack API request retries

All the message sends that result in a send to SlackMessageVerticle:

* SlackMessageVerticle is called by:
  - FinalizeJobVerticle
* FinalizeJobVerticle is called by:
  - BatchJobStatusHandler
  - LoadCsvHandler
  - ItemFailureVerticle
* ItemFailureVerticle is called by:
  - LoadCsvHandler

All the message sends to the above Verticles (amounting to five call
sites) now accommodate the Slack API request retries.

Commit:
Make variable name shorter

Commit:
Use slightly clearer logic
